### PR TITLE
Examples for RAG Benchmarking - LLM as a judge

### PR DESCRIPTION
**Pull request by sreanikdk**
This PR adds examples for 'RAG Benchmarking - LLM as a judge', as a new section.

### CODE REVIEW

**Review by SeriousSem — APPROVED** [review]

File: scripts/step03_explore_examples/06_rag_benchmarking/src/evaluate_with_golden_testset.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import logging
+import sys
+import csv
+from pathlib import Path
+
+from rag_benchmark_utils.common_utils import create_retriever, validate_response, print_results
+from helpers.config import TERRAFORM_DOCS_TABLE_NAME
+
+RELATIVE_FILE_PATH = Path("data/golden_test_set.csv")
```
Comment (romaniam) [inline scripts/step03_explore_examples/06_rag_benchmarking/src/evaluate_with_golden_testset.py:9]: Not a must but might be easier to configure if moved to the config file

File: scripts/step03_explore_examples/06_rag_benchmarking/src/evaluate_without_golden_testset.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    qa_chain = create_retriever(TERRAFORM_DOCS_TABLE_NAME)
+    results = []
+
+    for case in test_cases:
+        query = case["query"]
+
+        try:
+            log.info(f"Asking question: {query}")
+            result = qa_chain.invoke({"query": query})
+            actual_answer = result.get("result", "N/A")
```
Comment (romaniam) [inline scripts/step03_explore_examples/06_rag_benchmarking/src/evaluate_without_golden_testset.py:30]: Again not a must but constants for N/A and Failed would be better in this scenario

File: scripts/step03_explore_examples/06_rag_benchmarking/src/generate_golden_testset.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+from helpers.config import EMBEDDINGS_MODEL_NAME, CRITIC_LLM, TEST_SIZE, GENERATOR_LLM
+from utils.ingest_docs import fetch_terraform_docs
+
+GEN_AI_HUB = "gen-ai-hub"
+DISTRIBUTIONS = {
+    simple: 0.5,
+    multi_context: 0.4,
+    reasoning: 0.1
+}
+OUTPUT_PATH = Path('data/golden_test_set.csv')
```
Comment (romaniam) [inline scripts/step03_explore_examples/06_rag_benchmarking/src/generate_golden_testset.py:18]: Not a must but might be easier to configure if moved to the config file

**Review by romaniam — APPROVED** [review]
Some small remarks but in general LGTM

**Pull request merged**